BossMod/Modules/GoldSaucer/TheSliceIsRight/TheSliceIsRight.cs

```diff
@@ -0,0 +1,160 @@
+        Somethingoutsidethearena = 18338, // 2BC8->self, no cast, single-target
+        BambooSplit = 18333, // 25AD->self, 0.7s cast, range 28 width 5 rect
+        BambooCircleFall = 18334, // 25AD->self, 0.7s cast, range 11 circle 
+        BambooSpawn = 18327, // 25AD->self, no cast, range 3 circle
+        FirstGilJump = 18335, // 25AC->location, 2.5s cast, width 7 rect charge
+        NextGilJump = 18336, // 25AC->location, 1.5s cast, width 7 rect charge
+        BadCup = 18337, // 25AC->self, 1.0s cast, range 15+R 120-degree cone
+    };
+
+    class BambooSplits : Components.GenericAOEs
+    {
+        private List<Actor> _doublesidedsplit = new();
+        private List<Actor> _singlesplit = new();
+        private List<Actor> _circle = new();
+        private List<Actor> _doublesidedsplitToberemoved = new();
+        private List<Actor> _singlesplitToberemoved = new();
+        private List<Actor> _circleToberemoved = new();
+        private List<Actor> _bamboospawn = new();
+        private static readonly AOEShapeRect rectdouble = new(28, 2.5f, 28);
+        private static readonly AOEShapeRect rectsingle = new(28, 2.5f);
+        private static readonly AOEShapeCircle circle = new(11);
+        private static readonly AOEShapeCircle bamboospawn = new(3);
+        private DateTime _activation;
+        private DateTime _time;
+
+        public override IEnumerable<AOEInstance> ActiveAOEs(BossModule module, int slot, Actor actor)
+        {
+            foreach (var b in _doublesidedsplit)
+                yield return new(rectdouble, b.Position, b.Rotation + 90.Degrees(), _activation.AddSeconds(7));
+            foreach (var b in _singlesplit)
+                yield return new(rectsingle, b.Position, b.Rotation + 90.Degrees(), _activation.AddSeconds(7));
+            foreach (var b in _circle)
+                yield return new(circle, b.Position, b.Rotation, _activation.AddSeconds(7));
+            foreach (var b in _bamboospawn)
+                yield return new(bamboospawn, b.Position, b.Rotation); //activation time varies a lot (depending on the set?), just avoid entirely
+        }
+
+        public override void OnActorCreated(BossModule module, Actor actor)
+        {
+            if ((OID)actor.OID == OID.Bamboo && !_bamboospawn.Any(b => b.Position.AlmostEqual(actor.Position, 1)))
+                _bamboospawn.Add(actor);
+        }
+
+        public override void Update(BossModule module)
+        {
+            if (_time != default && module.WorldState.CurrentTime > _time)
+            {
+                _time = default;
+                _circle = _circle.Except(_circleToberemoved).ToList();
+                _circleToberemoved.Clear();
+                _singlesplit = _singlesplit.Except(_singlesplitToberemoved).ToList();
+                _singlesplitToberemoved.Clear();
+                _doublesidedsplit = _doublesidedsplit.Except(_doublesidedsplitToberemoved).ToList();
+                _doublesidedsplit.Clear();
```

Comment:
And this is just pointless - you first rebuild a list, and then immediately clear it?..